src/h3lib/lib/iterators.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2021 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @file IterCellsChildren.c
+ * @brief Iterator structs and functions for the children of a cell,
+ * or cells at a given resolution.
+ */
+
+#include "iterators.h"
+
+#include "h3Index.h"
+
+// extract the `res` digit (0--7) of the current cell
+static int _get(IterCellsChildren* it, int res) {
```

Comment:
Nit: `_getResDigit`?
Nit 2: Do we need the underscore prefix for functions marked `static`?

Reply:
changed.

I don't think we need the underscore, but it's a nice visual cue, i think. Also, it seems like we use the underscore throughout the rest of the library.